lib/handler/configurator/proxy.c

```diff
@@ -265,11 +266,131 @@ static int on_config_ssl_session_cache(h2o_configurator_command_t *cmd, h2o_conf
     return 0;
 }
 
+static int parse_balancer(h2o_configurator_command_t *cmd, yoml_t *node, h2o_socketpool_target_t **targets, size_t target_len)
+{
```

Comment:
Do we need to define this as a separate function now?

I believe that none of the balancers being defined take an argument (except the per-target weight). Therefore, the type of the `balancer` attribute could be restricted to `YOML_TYPE_SCALAR`. All we need to do is to call `strcasecmp` to determine the correct balancer constructor.